pkg/controller/core/workload_controller.go

```diff
@@ -252,3 +266,27 @@ func workloadFinished(w *kueue.Workload) bool {
 	}
 	return false
 }
+
+// We do not verify Pod's RuntimeClass legality here as this will be performed in admission controller.
+func handlePodOverhead(wl *kueue.Workload, classes nodev1.RuntimeClassList) {
+	for _, pod := range wl.Spec.PodSets {
+		if pod.Spec.RuntimeClassName != nil && len(pod.Spec.Overhead) == 0 {
+			for _, item := range classes.Items {
+				if item.Name == *pod.Spec.RuntimeClassName && item.Overhead != nil {
+					pod.Spec.Overhead = item.Overhead.PodFixed
+					break
+				}
+			}
+		} else if pod.Spec.RuntimeClassName == nil && len(pod.Spec.Overhead) > 0 {
+			pod.Spec.Overhead = make(corev1.ResourceList)
```

Comment:
This is modifying the object from the informer Cache. Maybe consider doing it in `NewInfo`?

Reply:
Oh yes, you're right. In this case, pod creation will be rejected original, but now it seems legally.